### PR TITLE
[OPERATOR-272,273,277] 1.4.5 issues

### PR DESCRIPTION
**Pull request by pure-jliao**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: drivers/storage/portworx/portworx.go [hunk 1]
```diff
@@ -685,7 +685,7 @@ func setDefaultAutopilotProviders(
 				Name: "default",
 				Type: "prometheus",
 				Params: map[string]string{
-					"url": "http://prometheus:9090",
+					"url": "http://px-prometheus:9090",
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:688]: Looks like we never had a unit test for this piece of code. Can we add a UT in this method - https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/portworx_test.go#L643?
Just check that there is a provider if you did not specify anything and it is equal to what we have here.

File: drivers/storage/portworx/portworx_test.go [hunk 2]
```diff
@@ -641,6 +641,12 @@ func TestStorageClusterDefaultsForAutopilot(t *testing.T) {
 	cluster.Spec.Autopilot.Enabled = false
 	driver.SetDefaultsOnStorageCluster(cluster)
 	require.Empty(t, cluster.Status.DesiredImages.Autopilot)
+
+	// Check default autopilot provider is set if not specified
+	driver.SetDefaultsOnStorageCluster(cluster)
+	providers := cluster.Spec.Autopilot.Providers
+	require.Equal(t, 1, len(providers))
+	require.Equal(t, component.AutopilotDefaultProviderEndpoint, providers[0].Params["url"])
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx_test.go:649]: Can we also validate the type of the provider?
```
require.Equal(t, "prometheus", providers[0].Type)
```

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM 
Feel free to merge after adding the assertion.

**Review by piyush-nimbalkar — APPROVED** [review]

**Review by haibinxie — APPROVED** [review]
LGTM

**Pull request merged**